metadata-io/src/test/java/com/linkedin/metadata/timeline/eventgenerator/SchemaMetadataChangeEventGeneratorTest.java

```diff
@@ -7,16 +7,128 @@
+import com.linkedin.schema.SchemaField;
+import com.linkedin.schema.SchemaFieldArray;
 import com.linkedin.schema.SchemaMetadata;
+import java.net.URISyntaxException;
 import java.nio.charset.StandardCharsets;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
+import java.util.Set;
 import org.apache.commons.io.IOUtils;
 import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
 import org.testng.annotations.Test;
 
 public class SchemaMetadataChangeEventGeneratorTest extends AbstractTestNGSpringContextTests {
 
+  private static Urn getTestUrn() throws URISyntaxException {
+    return Urn.createFromString("urn:li:dataset:(urn:li:dataPlatform:hdfs,SampleHdfsDataset,PROD)");
+  }
+
+  private static AuditStamp getTestAuditStamp() throws URISyntaxException {
+    return new AuditStamp()
+        .setActor(Urn.createFromString("urn:li:corpuser:__datahub_system"))
+        .setTime(1683829509553L);
+  }
+
+  private static void compareDescriptions(
+      Set<String> expectedDescriptions, List<ChangeEvent> actual) {
+    Set<String> actualDescriptions = new HashSet<>();
+    actual.forEach(
+        changeEvent -> {
+          actualDescriptions.add(changeEvent.getDescription());
+        });
+    assertEquals(expectedDescriptions, actualDescriptions);
+  }
+
+  private static Aspect<SchemaMetadata> getSchemaMetadata(List<SchemaField> schemaFieldList) {
+    return new Aspect<>(
+        new SchemaMetadata().setFields(new SchemaFieldArray(schemaFieldList)),
+        new SystemMetadata());
+  }
+
+  @Test
+  public void testNativeSchemaBackwardIncompatibleChange() throws Exception {
+    SchemaMetadataChangeEventGenerator test = new SchemaMetadataChangeEventGenerator();
+
+    Urn urn = getTestUrn();
+    String entity = "dataset";
+    String aspect = "schemaMetadata";
+    AuditStamp auditStamp = getTestAuditStamp();
+
+    Aspect<SchemaMetadata> from =
+        getSchemaMetadata(
+            List.of(new SchemaField().setFieldPath("ID").setNativeDataType("NUMBER(16,1)")));
+    Aspect<SchemaMetadata> to =
+        getSchemaMetadata(
+            List.of(new SchemaField().setFieldPath("ID").setNativeDataType("NUMBER(10,1)")));
+    List<ChangeEvent> actual = test.getChangeEvents(urn, entity, aspect, from, to, auditStamp);
+    // Test single field going from NUMBER(16,1) -> NUMBER(10,1)
+    assertEquals(1, actual.size());
+    compareDescriptions(
+        Set.of(
+            "A backwards incompatible change due to native datatype of the field 'ID' changed from 'NUMBER(16,1)' to 'NUMBER(10,1)'."),
+        actual);
+
+    List<ChangeEvent> actual2 = test.getChangeEvents(urn, entity, aspect, to, from, auditStamp);
+    // Test single field going from NUMBER(10,1) -> NUMBER(16,1)
+    assertEquals(1, actual2.size());
+    compareDescriptions(
+        Set.of(
+            "A backwards incompatible change due to native datatype of the field 'ID' changed from 'NUMBER(10,1)' to 'NUMBER(16,1)'."),
```

Comment:
I would like to see a test that checks for null-ability changes to a SchemaField. If a field goes from `nullable=false` to `nullable=true` it should be backwards incompatible but the inverse is not true

Reply:
We can take that as a follow up as current docs don't have this. But yes makes sense to enhance the API as per the demand from orgs. This is a bugfix of what is documented vs. what was working.